### ISSUE TITLE
UI Support Pluggable Drivers

### ISSUE THREAD

**Issue by nathan-jenan-rancher** [rancher/rancher#15101]
Add UI support for https://github.com/rancher/rancher/issues/12698

There will be an example dynamic driver to build the UI against (TBD ask @nathan-jenan-rancher ).

Edit for QA: See [below](https://github.com/rancher/rancher/issues/15101#issuecomment-443367631) note below about verification

**Comment by westlywright** [comment]
##QA

This PR adds a new page to `Custom Drivers -> Cluster Drivers` Route.

**Comment by davidnuzik** [comment]
@westlywright I'm not able to get step #1 of validation to work I tried the https://github.com/rancher/kontainer-engine-example-driver/releases/download/v0.1.0/kontainer-engine-example-driver-linux URL. A new release happened so changed that to v0.2.0 and still no joy. UI will stay in "Downloading" state and in the rancher server container logs I see a backend issue in the KontainerDriverController: 
![image](https://user-images.githubusercontent.com/45179589/50851050-ea35c700-1338-11e9-8bd8-1bca5c8d37ae.png)

```
2019/01/08 18:29:28 [INFO] create kontainerdriver kd-swmbf
2019/01/08 18:29:28 [ERROR] Returning previous error: failed to extract
2019/01/08 18:29:28 [ERROR] KontainerDriverController kd-swmbf recovered from panic "runtime error: invalid memory address or nil pointer dereference". (err=<nil>) Call stack:                                                                                              
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:154
/usr/local/go/src/runtime/asm_amd64.s:522
/usr/local/go/src/runtime/panic.go:513
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_kontainer_driver_controller.go:212                                                                                                                              
/go/src/github.com/rancher/rancher/pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go:79
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_kontainer_driver_lifecycle_adapter.go:29                                                                                                                        
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:219
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:191
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:181
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:219
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:64
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/lifecycle/object.go:45
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_kontainer_driver_lifecycle_adapter.go:56                                                                                                                        
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_kontainer_driver_controller.go:133                                                                                                                              
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:348
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:258
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:246
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:238
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1333
```

**Comment by davidnuzik** [comment]
Waiting on #17425 currently blocked by this.

**Comment by davidnuzik** [comment]
Version: master 2.2 (1/21/19)
No longer blocked by #17425 and I have **validated step 1 of 2** for the QA tests as indicated by Wes. I successfully deployed the example cluster driver. I validated the initial HTTP POST and edits (PUTs) and DELETE and validated the API per the instructions. Further, I reviewed the API schema to make sure it looks correct. There were no backend issues getting the custom driver and deploying it. I will connect with @westlywright regarding step 2 of QA validation.

**Comment by davidnuzik** [comment]
Version: master 2.2 (1/21/19)
I have **validated step 2 of 2** for the QA tests as indicated by Wes. I successfully built and started the example ui skeleton via https://github.com/rancher/ui-cluster-driver-skel/tree/google-gke-example
In the UI I was able to get this to work without issue; I validated HTTP and using the example I was able to provision a GKE cluster via the UI. Validated edit cluster and delete cluster as well. I inspected the API as well and did not identify any problems. The cluster was smoke tested.

While the delete function worked there was a new bug found around this entered here: #17495

Screenshot from the "Add Cluster" page:
![image](https://user-images.githubusercontent.com/45179589/51492335-52879e00-1d66-11e9-8098-15359fe7ebc4.png)
